src/runtime_src/core/include/unittests/xrt_h_c_compilation/CMakeLists.txt

```
@@ -0,0 +1,4 @@
+# SPDX-License-Identifier: Apache-2.0                                           
+# Copyright (C) 2022, Advanced Micro Devices, Inc. All rights reserved.
+
+add_executable(main main.c)
```

```suggestion
add_library(unit_cpps OBJECT main.c)
```
You only care about compiling the object file, not linking the executable, so I think this will do?